deps.bzl

```diff
@@ -6011,8 +6023,8 @@ def install_go_mod_dependencies(workspace_name = "buildbuddy"):
     go_repository(
         name = "in_gopkg_square_go_jose_v2",
         importpath = "gopkg.in/square/go-jose.v2",
-        sum = "h1:NGk74WTnPKBNUhNzQX7PYcTLUjoq7mzKk2OKbvwk2iI=",
-        version = "v2.6.0",
+        sum = "h1:7odma5RETjNHWJnR32wx8t+Io4djHE1PqxCFx3iiZ2w=",
+        version = "v2.5.1",
     )
```

Comment:
This is a bit puzzling to me.

AFAIK, we are not using this anymore in our build graph. However, it's still listed as one of our transitive dependencies in `go.sum`. And because Gazelle uses `go.sum` to generate `deps.bzl`, we still have it declared here.

cc: @fmeum on the Gazelle behavior.

Reply:
`go_deps` goes by `go.mod`, which has package-level granularity. I don't know how we would fix this for `update-repos`, but since we want to phase it out anyway, we should probably just accept this oddness.

Reply:
```
> go mod graph | grep 'jose.v2' | sort
github.com/containerd/containerd@v1.6.8 gopkg.in/square/go-jose.v2@v2.5.1
github.com/containerd/containerd@v1.7.14 gopkg.in/square/go-jose.v2@v2.5.1
github.com/containers/ocicrypt@v1.0.1 gopkg.in/square/go-jose.v2@v2.3.1
github.com/containers/ocicrypt@v1.1.0 gopkg.in/square/go-jose.v2@v2.5.1
github.com/containers/ocicrypt@v1.1.1 gopkg.in/square/go-jose.v2@v2.5.1
github.com/containers/ocicrypt@v1.1.2 gopkg.in/square/go-jose.v2@v2.5.1
github.com/containers/ocicrypt@v1.1.3 gopkg.in/square/go-jose.v2@v2.5.1
github.com/google/tink/go@v1.7.0 gopkg.in/square/go-jose.v2@v2.5.1
k8s.io/apiserver@v0.20.1 gopkg.in/square/go-jose.v2@v2.2.2
k8s.io/apiserver@v0.20.4 gopkg.in/square/go-jose.v2@v2.2.2
k8s.io/apiserver@v0.20.6 gopkg.in/square/go-jose.v2@v2.2.2
k8s.io/apiserver@v0.22.5 gopkg.in/square/go-jose.v2@v2.2.2
```

So here is why we still need jose.v2. All are irrelevant to our build.

As for the current `update-repos` behavior:

Gazelle is calling this command to get the list of modules https://github.com/bazelbuild/bazel-gazelle/blob/852fdcfe7d6cdfcc72b1275a2200a1e131bf0203/language/go/utils.go#L36-L39.
It's unclear to me if there is another command that we could use with similar outputs, but filter out all the transitive deps that are not used.